src/app/components/add-card/add-card.component.ts

```diff
@@ -81,6 +82,17 @@ export class AddCardComponent implements OnDestroy {
     this.newTask = '';
   }
 
+  onTaskButtonClick(event: {id: number, dataJob: string}): void {
+    const id = event.id;
+    const dataJob = event.dataJob;
+    const taskIndex = this.taskService.taskList.findIndex(task => task.id === id);
+    if (taskIndex !== -1 && dataJob === this.constantsService.COMPLETE ) {}
+    else if (dataJob === this.constantsService.EDIT) {} 
+    else if (dataJob === this.constantsService.DELETE_TODO) {
+      this.taskService.taskList.splice(taskIndex, 1);
+    } else {}
+  }
```

Comment:
```suggestion
  onTaskButtonClick({ id: number, dataJob: string }): void {
  }
```

I would destructure like above.

Also I am not sure why the if/else/elseif blocks are empty. Please clean up the code and keep only what necessary.

Reply:
cleaned up done!